data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/informationtype/FilterRequest.java

```diff
@@ -0,0 +1,149 @@
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import no.nav.data.catalog.backend.app.codelist.ListName;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+
+@Data
+@NoArgsConstructor
+public class FilterRequest {
+	private Specification<InformationType> specification;
+	private Pageable pageable;
+
+	public FilterRequest mapFromQuery(Map<String, String> queryMap) {
+
+		if (queryMap.isEmpty()) {
+			this.specification = null;
+			this.pageable = PageRequest.of(0, 20);
+			return this;
+		}
+		checkForAndSetPageable(queryMap);
+		checkForAndSetSpecification(queryMap);
+		return this;
+	}
+
+	private void checkForAndSetPageable(Map<String, String> queryMap) {
+		int page = queryMap.get("page") == null ? 0 : Integer.parseInt(queryMap.get("page"));
+		int pageSize = queryMap.get("pageSize") == null ? 20 : Integer.parseInt(queryMap.get("pageSize"));
+
+		if (queryMap.get("sort") != null) {
+			Queue<String> queue = new LinkedList<>(Arrays.asList(queryMap.get("sort").split(",")));
+			Sort sort = getSortQuery(queue);
+			this.pageable = PageRequest.of(page, pageSize, sort);
+		} else {
+			this.pageable = PageRequest.of(page, pageSize);
+		}
+	}
+
+	private Sort getSortQuery(Queue<String> queue) {
+		String current = queue.remove();
+		String next = queue.peek();
+
+		if (next == null) {
+			if (current.equalsIgnoreCase("asc")) {
+				return Sort.unsorted();
+			}
+			if (current.equalsIgnoreCase("desc")) {
+				return Sort.unsorted().descending();
+			}
+			return Sort.by(current);
+		} else if (next.equalsIgnoreCase("asc")) {
+			queue.remove(); // removes direction from queue
+			next = queue.peek();
+			if (next == null) {  // check if end of queue
+				return Sort.by(current).ascending();
+			}
+			return Sort.by(current).ascending().and(getSortQuery(queue));
+		} else if (next.equalsIgnoreCase("desc")) {
+			queue.remove(); // removes direction from queue
+			next = queue.peek();
+			if (next == null) {  // check if end of queue
+				return Sort.by(current).descending();
+			}
+			return Sort.by(current).descending().and(getSortQuery(queue));
+		} else {
+			return Sort.by(current).and(getSortQuery(queue));
+		}
+	}
+
+	private void checkForAndSetSpecification(Map<String, String> queryMap) {
+		List<Specification<InformationType>> specificationList = new ArrayList<>();
+
+		if (queryMap.get("name") != null) {
+			specificationList.add(InformationTypeSpecification.hasName(queryMap.get("name")));
+		}
+		if (queryMap.get("description") != null) {
+			specificationList.add(InformationTypeSpecification.hasDescription(queryMap.get("description")));
+		}
+		if (queryMap.get("personalData") != null) {
```

Comment:
Kan kanskje legge på en kort sjekk av om man sender inn en av de to lovlige verdiene: true eller false